code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,494 @@
+							/datum/xenoartifact_trait/activator/signal,/datum/xenoartifact_trait/major/heal,
+							/datum/xenoartifact_trait/activator/signal, /datum/xenoartifact_trait/activator/batteryneed))
+			if(!xenop.price)
+				xenop.price = pick(200, 300, 500)
+			malfunction_mod = 2
+
+		if(URANIUM)
+			name = "uranium [name]"
+			generate_traits(list(/datum/xenoartifact_trait/major/sing, /datum/xenoartifact_trait/minor/sharp,
+							/datum/xenoartifact_trait/major/laser, /datum/xenoartifact_trait/major/corginator,
+							/datum/xenoartifact_trait/minor/sentient, /datum/xenoartifact_trait/minor/wearable,
+							/datum/xenoartifact_trait/major/handmore, /datum/xenoartifact_trait/major/invisible,
+							/datum/xenoartifact_trait/major/heal), TRUE) 
+			if(!xenop.price)
+				xenop.price = pick(300, 500, 800) 
+			malfunction_mod = 8
+
+		if(BANANIUM)
+			name = "bananium [name]"
+			generate_traits(list(/datum/xenoartifact_trait/major/sing))
+			if(!xenop.price)
+				xenop.price = pick(500, 800, 1000) 
+			malfunction_mod = 0.5
+
+	icon_state = null
+	for(var/datum/xenoartifact_trait/T in traits) //This is kinda weird but it stops certain bugs.
+		if(istype(T, /datum/xenoartifact_trait/minor/dense))
+			T.on_init(src)
+			return
+	for(var/datum/xenoartifact_trait/T in traits)
+		T.on_init(src)
+
+	//Random sprite process, I'd like to maybe revisit this, make it a function. probably don't
+	if(!(icon_state))
+		var/holdthisplease = rand(1, 3)
+		icon_state = "IB[holdthisplease]"//base
+		generate_icon(icon, "IBL[holdthisplease]", material)
+	if(prob(50) || icon_slots[1])//Top
+		if(!(icon_slots[1])) //Some traits can set this too, it will be set to a code that looks like 901, 908, 905 ect.
+			icon_slots[1] = rand(1, 2)
+		generate_icon(icon, "ITP[icon_slots[1]]")
+		generate_icon(icon, "ITPL[icon_slots[1]]", material)
+
+	if(prob(30) || icon_slots[3])//Left
+		if(!(icon_slots[3]))
+			icon_slots[3] = rand(1, 2)
+		generate_icon(icon, "IL[icon_slots[3]]")
+		generate_icon(icon, "ILL[icon_slots[3]]", material)
+
+	if(prob(50)  || icon_slots[4])//Right
+		if(!(icon_slots[4]))
+			icon_slots[4] = rand(1, 2)
+		generate_icon(icon, "IR[icon_slots[4]]")
+		generate_icon(icon, "IRL[icon_slots[4]]", material)
+
+	if(prob(30) || icon_slots[2])//Bottom
+		if(!(icon_slots[2]))
+			icon_slots[2] = rand(1, 2)
+		generate_icon(icon, "IBTM[icon_slots[2]]")
+		generate_icon(icon, "IBTML[icon_slots[2]]", material)
+
+/obj/item/xenoartifact/examine(mob/living/carbon/user)
+	. = ..()
+	if(istype(user?.glasses, /obj/item/clothing/glasses/science))
+		to_chat(user, "<span class='notice'>[special_desc]</span>")
+
+/obj/item/xenoartifact/interact(mob/user)
+	. = ..()
+	if(process_type == LIT) //Snuff out candle
+		to_chat(user, "<span class='notice'>You snuff out [name]</span>")
+		process_type = null
+		return
+	if(user.a_intent == INTENT_GRAB)
+		touch_desc?.on_touch(src, user)
+		return
+	SEND_SIGNAL(src, XENOA_INTERACT, null, user, user)
```

Comment:
The old signals still get sent. These signals are used specifically because they all send data in same order, I don't think the other signals do. Using this method you can define which xeno-signals activate your activator without having to write a new interaction for every signal and define new traits more effectively.

Reply:
Reuse the already sent signal then